router/src/main/java/org/simple/clinic/router/screen/FullScreenKeyChanger.kt

```diff
@@ -43,7 +45,12 @@ class FullScreenKeyChanger(
     // end up overlapping with each other. It's difficult to debug if it's a problem with Flow
     // or in our code. As a workaround, the window background is applied on every screen.
     val container = FrameLayout(incomingContext)
-    container.id = View.generateViewId()
+
+    // The ID for each screen's container should remain the same for View state restoration to work.
+    if (containerIds.containsKey(incomingKey).not()) {
+      containerIds += incomingKey to View.generateViewId()
```

Comment:
I missed this: `+=` has different behaviour depending on whether its called on a `MutableMap` (adds the entry to the map) vs a `Map` (returns a new `Map` with the older entries and the newly added one).

I believe we should use `+=` only when creating a new map, and not modifying a `MutableMap`. This will also make our usage of the map operations consistent (in `removeOutgoingView`).